### PR TITLE
mehmetEkinci-homework-week2

### PR DESCRIPTION
**Pull request by mehmetekinci**

### CODE REVIEW

File: week2/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,117 @@
...
+const filePath = './list.json';
+const defaultEncoding = 'utf-8';
+
+const writeFilePromisify = promisify(fs.writeFile);
+const appendFilePromisify = promisify(fs.appendFile);
+const readFilePromisify = promisify(fs.readFile);
+
+let todos;
+
+const getAllTodos = async () => {
```
Comment (gajduk) [inline week2/homework/src/index.js]: 👍 nice to make this a separate function

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,117 @@
...
+  await getAllTodos();
+  const updatedTodos = todos.map(todo => {
+    if (todo.id === id) {
+      return {
+        id,
+        title: newTitle,
+      };
+    }
+    return todo;
+  });
```
Comment (gajduk) [inline week2/homework/src/index.js]: Your code is correct but I think the following is simpler and easier to understand

```javascript
let toUpdate = todos.find(todo => todo.id === ud)
toUpdate.title = newTitle;
```

File: week2/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,117 @@
...
+  await writeFilePromisify(filePath, writeData);
+};
+
+const removeTodo = async id => {
+  try {
+    await getAllTodos();
+    const filteredTodos = todos.filter(todo => todo.id !== id);
+    const writeData = JSON.stringify({ todos: filteredTodos }, null, 2);
+    await writeFilePromisify(filePath, writeData);
+  } catch (err) {
```
Comment (gajduk) [inline week2/homework/src/index.js]: what possible error are you catching here?

File: week2/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,117 @@
...
+
+const [, , ...restArguments] = process.argv;
+
+switch (restArguments[0]) {
+  case 'list':
+    getAllTodos().then(console.log);
+    break;
+  case 'add':
+    addTodo(parseInt(restArguments[1], 10), restArguments[2]).then(
+      console.log(`you successfully added a new todo`),
```
Comment (gajduk) [inline week2/homework/src/index.js]: 👍 nice to provide some feedback to the user

File: week2/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,117 @@
...
+    removeTodo(parseInt(restArguments[1], 10)).then(
+      console.log(`you successfully removed the todo`),
+    );
+    break;
+  case 'reset':
+    createEmptyJSONFile().then(console.log(`you successfully reset`));
+    break;
+  case 'help':
+    createHelpOptions();
+    break;
```
Comment (gajduk) [inline week2/homework/src/index.js]: 👍 very clean

File: week2/homework/src/index.js (hunk 2 above)
Comment (mehmetekinci) [reply week2/homework/src/index.js]: hey @gajduk . i tried way of you but I had a problem. toUpdate(it is the new todo) shows only it self. but my updateTodos shows all todos including new todo. how can I handle it?

File: week2/homework/src/index.js (hunk 3 above)
Comment (mehmetekinci) [reply week2/homework/src/index.js]: yes,it catch nothing. thanks for making me think.

**Pull request closed without merging**